Lib/idlelib/idle_test/test_pyparse.py

```diff
@@ -149,6 +149,10 @@ def test_set_lo(self):
         p.set_lo(44)
         self.assertEqual(p.str, code[44:])
 
+    def test_tran(self):
+        self.assertEqual('\t a([{b}])b"c\'d\n'.translate(self.parser._tran),
+                          'xxx(((x)))x"x\'x\n')
```

Comment:
I was wondering if explicit tests should be added for that.  I don't know if compressing a string like this is common so that it can be parsed faster, but I learned a lot from this technique.

Reply:
Abstractly, mappings implement functions (in the set-theoretic sense).  Copying the code that produces the mapping is not terribly useful, but in this case, I discovered a single 'actual = expected' test.